cmd/execution_builder.go

```diff
@@ -699,17 +724,51 @@ func (exeNode *ExecutionNode) LoadExecutionState(
 	error,
 ) {
 
-	chunkDataPackDB, err := openChunkDataPackDB(exeNode.exeConf.chunkDataPackDir, node.Logger)
+	// in order to be backward-compatible, we check whether the folder
+	// has been used by either badger or pebble database,
+	// and initialize the database accordingly.
+	isPebble, isBadger, isEmpty, err := storageerr.CheckFolder(exeNode.exeConf.chunkDataPackDir)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not check chunk data pack directory: %w", err)
+	}
+
+	if !isPebble && !isBadger && !isEmpty {
+		return nil, fmt.Errorf("chunk data pack directory is not a badger or pebble database: %s", exeNode.exeConf.chunkDataPackDir)
+	}
+
+	var closer io.Closer
+	var chunkDataPacks storageerr.ChunkDataPacks
+	node.Logger.Info().
+		Bool("is_pebble", isPebble).
+		Bool("is_badger", isBadger).
+		Bool("is_empty", isEmpty).
+		Str("chunk_data_pack_dir", exeNode.exeConf.chunkDataPackDir).
+		Msgf("opening chunk data pack database")
+
+	if isBadger {
```

Comment:
This part is the new change that only for v0.33 to be backward compatible, which [this PR does not have](https://github.com/onflow/flow-go/pull/6042)